### PR TITLE
Make ResourceLimitPolicy functions public to be used from scheduler plugins.

### PR DESCRIPTION
**Pull request by vvivekiyer**
Change is required for Schedulers that are built using plugins.

### CODE REVIEW

**Review by Jackie-Jiang — APPROVED** [review]
Out of the scope of this PR, can you share how do you plan to use this policy?
This module is never enabled or verified (`PrioritySchedulerTest` is disabled because of known race condition), and I believe this project was not complete. We might want to redesign it since currently the only supported one is FCFS

**Comment by vvivekiyer** [comment]
@Jackie-Jiang 
We are not using the PriorityBasedSchedulers. Just curious - can you point me to the race condition?

About this change -  We have a use case in Linkedin where we want to query production tables using adhoc channels (tools, scripts, reports, etc). Note that these adhoc queries are not from the site-facing applications but still need to be executed on production tables with some reasonably degraded latency. 

So we've built an internal scheduler (short term) that caps the amount of CPU resources given to these adhoc workloads while still letting the regular production workloads perform as usual. This is just a short term solution.

For the long term, I'm planning to leverage the above idea to build a long-term policy based workload management solution. I'm writing a PEP document for the overall effort and I'll share it with OSS soon (week or two).

**Review by siddharthteotia — APPROVED** [review]

**Comment by Jackie-Jiang** [comment]
@vvivekiyer I don't know the exact race condition, but the test in `PrioritySchedulerTest` was disabled due to race condition. Use this class with cautious because it is not verified in any production environment

**Comment by siddharthteotia** [comment]
@Jackie-Jiang  - @vvivekiyer  and I will provide you with the context. Will setup something soon.

TLDR - We have started thinking about the scheduling sub-problem described in https://github.com/apache/pinot/issues/8618 and https://github.com/apache/pinot/issues/9473 and have built a very limited scope solution that has shown good results for a production mixed workload. 

@vvivekiyer will be sharing a PEP
